### ISSUE TITLE
Docker image aliases for major version

### ISSUE THREAD

**Issue by laupow** [CircleCI-Public/cimg-node#130]
Can we get more aliases for the major releases?

Specifically these docker tags on `cimg/node`:
* `10` 
* `12`
* `14`
* `15`

Otherwise our CI pipelines get stale too quick as developers hardcode minor and patch versions in the pipeline config 😢

**Comment by felicianotech** [comment]
Hey all,

Thank you for the Issue. We've gotten this request before. Normally, I would tell you it's coming, link off the to feature request, and close this. I think this request is especially desired by Node.js users so I will leave this Issue open until it is completed.

To get to the point, this will happen in the near future though I don't current'y have a date. You can track this request here on this GitHub Issue or the [CircleCI Feature Request](https://ideas.circleci.com/images/p/support-major-version-tags) I created for this. You can upvote there to help push this along.

---

I do want to provide a little info as to why this isn't complete yet and why PR #132 wasn't merged. For the next-gen Convenience Images (the images on Docker Hub that are prefixed with `cimg/`, we have a shared build system for all of the images. This shared build system is used to build all Dockerfiles as well as the tags that each image will get.

This system do not yet support major version only tags. There's a few version for this:

1. In the past, users using this kind of tag, getting a larger update that they expected (from the software upstream, not us) and then blaming us for the breaking change. Limiting to minor versions so far has helped us mitigate some of these problems.
2. We need to occasionally change software in an image. For the Node.js image for example, it's not just the Node.js version that we pay attention to. There's npm, Yarn, Docker, git, and the Ubuntu base image itself. We don't make these changes to an image once it's tag has been published for stability, but we have to eventually. We try to do these within major versions as to limit the interruptions. Having a major version only tag, as requested, increases the likelihood that a future build of yours may break without any changes from you. This will be a trade off that we want to make extremely clear and upfront before we support major version only tags.
3. There's other related work that needs to happen to our build system regarding versions. For example, there's the effort to add [beta and RC releases](https://ideas.circleci.com/images/p/support-beta-and-rc-versions) to Convenience Images. This requires more complex logic to support beta and release candidate images for a particular major version, while still keeping to alias pointing only as stable releases.

To sum up, support is coming for this, and the Node.js image will be the first image to get this support when it comes.

**Comment by felicianotech** [comment]
Closing this for now but please read my previous comment.

**Comment by glasser** [comment]
@felicianotech Just to check — a year ago you said it was coming soon, and last month you closed it referring to the previous comment. Does that mean it is still coming soon? This would be super helpful.

**Comment by laupow** [comment]
It may not be mentioned as prominently as it should be in CircleCI docs, but the Node orb can install any version that `nvm` accepts. I've used this workaround a couple of time.

So a pipeline like this may accomplish your goals:
```
version: 2.1

executors:
  linux:
    docker:
      - image: cimg/base:stable

orbs:
  node: circleci/node@5

jobs:
  run-tests:
    parameters:
      node-version:
        type: string
    executor: linux
    steps:
      - checkout
      - node/install:
          node-version: << parameters.node-version >>
      - node/install-packages
      - run:
          name: Run tests
          command: npm run test:ci

workflows:
  version: 2
  test:
    jobs:
      - run-tests:
          matrix:
            parameters:
              node-version: [ "12", "14", "16", "latest"]

```
(I didn't test this)

For Windows/OSX jobs I think nvm is still present. The general idea works, but you might not be able to use docker.
